### PR TITLE
Fix: rm Gnosis from app text

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves
There were a couple places in the app where Gnosis the company was mentioned. I reworded them.

### CODE REVIEW

File: src/components/create-safe/steps/SetNameStep.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const SetNameStep = ({ params, onSubmit, onBack, setStep }: Props) => {
           <Box padding={3}>
             <Typography variant="body1" mb={2}>
               You are about to create a new Safe wallet with one or more owners. First, let&apos;s give your new wallet
-              a name. This name is only stored locally and will never be shared with Gnosis or any third parties. The
+              a name. This name is only stored locally and will never be sent to the backend or any third parties. The
```
Comment (iamacook) [inline src/components/create-safe/steps/SetNameStep.tsx:55]: I think "shared with Safe" or "shared with us" is a nicer phrasing. It's less technical.

Comment (iamacook) [reply src/components/create-safe/steps/SetNameStep.tsx:55]: Same goes for `<SetAddressStep>`.

File: src/components/safe-apps/AddCustomAppModal.tsx [hunk 2]
```diff
@@ -65,7 +66,7 @@ const AddCustomAppModal = ({ open, onClose, onSave, safeAppsList }: Props) => {
     if (!isValidURL(debouncedUrl)) return
 
     return fetchSafeAppFromManifest(debouncedUrl, chainId).catch(() => {
-      setError('appUrl', { type: 'custom', message: "The App doesn't support Safe App functionality" })
+      setError('appUrl', { type: 'custom', message: "The app doesn't support Safe App functionality" })
```
Comment (iamacook) [inline src/components/safe-apps/AddCustomAppModal.tsx:69]: Don't we regard Safe Apps as proper nouns?

Comment (iamacook) [reply src/components/safe-apps/AddCustomAppModal.tsx:69]: If they are, then line 133 would need adjusting.

File: src/components/create-safe/steps/SetNameStep.tsx (hunk 1 above)
Comment (iamacook) [reply src/components/create-safe/steps/SetNameStep.tsx:55]: And `<OwnerList>`.

File: src/components/safe-apps/AddCustomAppModal.tsx (hunk 2 above)
Comment (katspaugh) [reply src/components/safe-apps/AddCustomAppModal.tsx:69]: Safe Apps yes, but here it just says "this app isn't a Safe App"

**Review by iamacook — APPROVED** [review]

**Pull request merged**